code/clue.js

```diff
@@ -1,59 +1,460 @@
+  setTimeout(pickKiller, 2000);
+  revealedMystery.innerHTML = '';
+  killerCard.style.background = "";
+  killerName.innerHTML = '';
+  killerAge.innerHTML = '';
+  killerFavouriteWeapon.innerHTML = '';
+  killerImage.style.display = "none";
+  killerImage.src = '';
+};
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  mystery.favouriteWeapon = mystery.killer.favouriteWeapon;
+  killerLoader.style.display = "none";
+  killerCard.style.background = mystery.killer.color;
+  killerName.innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  killerAge.innerHTML = `Age: ${mystery.killer.age}`;
+  killerFavouriteWeapon.innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+  killerImage.style.display = "inline-block";
+  killerImage.src = mystery.killer.image;
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+killerCard.onclick = killerAnimation;
+
+// Actions related to Weapon deck
+
+const favouriteWeaponAnimation = () => {
+  if (mystery.killer === undefined) {
+    revealedMystery.innerHTML = ('Pick a killer first!');
+  } else {
+    weaponLoader.style.display = "block";
+    weaponName.innerHTML = '';
+    weaponWeight.innerHTML = '';
+    weaponImage.src = '';
+    weaponImage.style.display = "none";
+    weaponFavouriteRoom.innerHTML = '';
+    revealedMystery.innerHTML = '';
+    setTimeout(() => {
+      pickFavouriteWeapon(mystery.killer);
+    }, 2000);
+  };
+};
+
+const favouriteWeaponDuplicated = (elementToPopulate, howManyTimes) => {
+  let result = [];
+  for (let i = 0; i < howManyTimes; i++) {
+    result.push(elementToPopulate);
+  };
+  return result;
+};
+
+const pickFavouriteWeapon = (suspect) => {
+  const newArrayOfWeapons = weapons.map(a => a.name);
+  const randomWeapons = [
+    ...newArrayOfWeapons,
+    ...favouriteWeaponDuplicated(suspect.favouriteWeapon, 10)
+  ];
+  mystery.pickedWeapon = randomSelector(randomWeapons);
+  const mysteryWeapon = weapons.find((item) => {
+    if (item.name === mystery.pickedWeapon) {
+      return item;
+    };
+  });
+  mystery.weapon = mysteryWeapon;
+  mystery.favouriteRoom = mysteryWeapon.favouriteRoom;
+  weaponLoader.style.display = "none";
+  weaponCard.style.background = "#808080";
+  weaponName.innerHTML = mystery.weapon.name;
+  weaponWeight.innerHTML = `Weight: ${mystery.weapon.weight} oz`;
+  weaponImage.style.display = "inline-block";
```

Comment:
Consider adding a class instead of styling in the JS-file

Reply:
I created a new class `.weapon-deck-clicked` and replaced `weaponCard.style.background = "#808080";` by `weaponCard.classList.add('weapon-deck-clicked');`
Was it what you meant?

Reply:
Yeah, so you can have all the styling in the CSS and then just adding classes in JS 👍